src/window/concept_frame.rs

```diff
@@ -112,12 +112,33 @@ impl Part {
         parent: &wl_surface::WlSurface,
         compositor: &Attached<wl_compositor::WlCompositor>,
         subcompositor: &Attached<wl_subcompositor::WlSubcompositor>,
+        inner: Option<Rc<RefCell<Inner>>>,
     ) -> Part {
-        let surface = crate::surface::setup_surface(
-            compositor.create_surface(),
-            None::<fn(_, _, DispatchData)>,
-        );
+        let surface = if let Some(inner) = inner {
+            crate::surface::setup_surface(
+                compositor.create_surface(),
+                Some(
+                    move |dpi, surface: wl_surface::WlSurface, ddata: DispatchData| {
+                        surface.set_buffer_scale(dpi);
+                        surface.commit();
+                        (&mut inner.borrow_mut().implem)(FrameRequest::Refresh, 0, ddata);
+                    },
+                ),
+            )
+        } else {
+            crate::surface::setup_surface(
+                compositor.create_surface(),
+                Some(
+                    move |dpi, surface: wl_surface::WlSurface, _ddata: DispatchData| {
+                        surface.set_buffer_scale(dpi);
+                        surface.commit();
+                    },
+                ),
+            )
+        };
```

Comment:
I'm not sure if this is something desirable tbh. I'm not an expert in plumbing and sctk internals, so suggestions are welcome.

Reply:
I'm alright with that.